api/v1alpha4/types.go

```diff
@@ -130,6 +130,8 @@ type PortOpts struct {
 	ProjectID           string        `json:"projectId,omitempty"`
 	SecurityGroups      *[]string     `json:"securityGroups,omitempty"`
 	AllowedAddressPairs []AddressPair `json:"allowedAddressPairs,omitempty"`
+	// Enables and disables trunk at port level. Inhertis from the corresponding network level field, if not provided.
```

Comment:
just a nit. is this comment correct(needed?)  wouldn't "network level field" be openStackMachine.Spec.network.Trunk? 
But the port inherits openStackMachine.Spec.Trunk (https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/934/files#diff-5e7dbfd52305a21a11fb94fab201062ae2c7b851db93c2a8443a1d82d7aa7936R146)

Perhaps drop the comment to avoid confusion..

Reply:
True, will correct it. We need the comment is it is needed, visible when running `kubectl explain`
I am also working on PortSecurity, confused the two.